src/deepsparse/transformers/haystack.py

```diff
@@ -0,0 +1,359 @@
+                "dot_product instead. This can be set when initializing the "
+                "DocumentStore"
+            )
+        if pooling_strategy != "cls_token":
+            _LOGGER.warning(
+                "You are using a Dense Passage Retriever model with "
+                f"{pooling_strategy} pooling_strategy. We recommend you use "
+                "cls_token instead"
+            )
+        if pooling_strategy == "per_token" and max_seq_len_query != max_seq_len_passage:
+            raise ValueError(
+                "per_token pooling strategy requires that max_seq_len_query "
+                f"({max_seq_len_query}) match max_seq_len_passage "
+                f"({max_seq_len_passage})"
+            )
+        if embed_title:
+            raise ValueError(
+                "DeepSparseDensePassageRetriever does not support embedding titles"
+            )
+
+        if self.context is None:
+            self.context = Context(
+                num_cores=None, num_streams=4
+            )  # arbitrarily choose 4
+
+        _LOGGER.info("Creating query pipeline")
+        self.query_pipeline = Pipeline.create(
+            "embedding_extraction",
+            query_model_path,
+            batch_size=batch_size,
+            sequence_length=max_seq_len_query,
+            emb_extraction_layer=-1,
+            extraction_strategy=pooling_strategy,
+            show_progress_bar=progress_bar,
+            context=context,
+            **pipeline_kwargs,
+        )
+        _LOGGER.info("Creating passage pipeline")
+        self.passage_pipeline = Pipeline.create(
+            "embedding_extraction",
+            passage_model_path,
+            batch_size=batch_size,
+            sequence_length=max_seq_len_passage,
+            emb_extraction_layer=-1,
+            extraction_strategy=pooling_strategy,
+            show_progress_bar=progress_bar,
+            context=context,
+            **pipeline_kwargs,
+        )
+        _LOGGER.info("Query and passage pipelines initialized")
+
+    def embed_queries(self, texts: List[str]) -> List[numpy.ndarray]:
+        """
+        :param texts: list of query strings to embed
+        :return: list of embeddings for each query
+        """
+        return self.query_pipeline(texts).embeddings
+
+    def embed_documents(self, docs: List[Document]) -> List[numpy.ndarray]:
+        """
+        :param docs: list of document strings to embed
+        :return: list of embeddings for each document
+        """
+        passage_inputs = [doc.content for doc in docs]
+        return self.passage_pipeline(passage_inputs).embeddings
+
+    def _get_predictions(*args, **kwargs):
+        raise NotImplementedError()
+
+    def train(*args, **kwargs):
+        raise NotImplementedError()
+
+    def save(*args, **kwargs):
+        raise NotImplementedError()
+
+    def load(*args, **kwargs):
+        raise NotImplementedError()
+
+
+class _DeepSparseEmbeddingEncoder(_BaseEmbeddingEncoder):
```

Comment:
why is this private?

Reply:
This class has no use outside of the `DeepSparseEmbeddingRetriever`, so there is no reason why this class would be needed to be used directly.

Reply:
After convo decided to expose publicly